File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/util/CompressionTypeHelper.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3.util;

import static io.airbyte.integrations.destination.s3.S3DestinationConstants.COMPRESSION_TYPE_ARG_NAME;
import static io.airbyte.integrations.destination.s3.S3DestinationConstants.DEFAULT_COMPRESSION_TYPE;

import com.fasterxml.jackson.databind.JsonNode;

public class CompressionTypeHelper {

  private CompressionTypeHelper() {}

  /**
   * Sample expected input: { "compression_type": "No Compression" }
   */
  public static CompressionType parseCompressionType(final JsonNode compressionConfig) {
    if (compressionConfig == null || compressionConfig.isNull()) {
      return DEFAULT_COMPRESSION_TYPE;
    }
    final String compressionType = compressionConfig.get(COMPRESSION_TYPE_ARG_NAME).asText();
    if (compressionType.toUpperCase().equals(CompressionType.GZIP.name())) {
      return CompressionType.GZIP;
    } else {
      return CompressionType.NO_COMPRESSION;
    }
  }

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/util/CompressionTypeHelper.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.s3.util

import com.fasterxml.jackson.databind.JsonNode
import io.airbyte.cdk.integrations.destination.s3.S3DestinationConstants
import java.util.*

object CompressionTypeHelper {
    /** Sample expected input: { "compression_type": "No Compression" } */
    @JvmStatic
    fun parseCompressionType(compressionConfig: JsonNode?): CompressionType {
        if (compressionConfig == null || compressionConfig.isNull) {
            return S3DestinationConstants.DEFAULT_COMPRESSION_TYPE
        }
        val compressionType =
            compressionConfig[S3DestinationConstants.COMPRESSION_TYPE_ARG_NAME].asText()
        return if (compressionType.uppercase(Locale.getDefault()) == CompressionType.GZIP.name) {
            CompressionType.GZIP
        } else {
            CompressionType.NO_COMPRESSION
        }
    }
}
